code/modules/materials/materials.dm

```diff
@@ -55,12 +55,28 @@ var/list/name_to_material
 		populate_material_list()
 	return name_to_material[name]
 
+/proc/material_by_stack_type(stype)
```

Comment:
`get_material_name_by_stack_type` probably better name.
`material_by_stack_type` can be confusing.